openbas-api/src/main/java/io/openbas/rest/team/TeamApi.java

```diff
@@ -236,4 +232,26 @@ public Team updateTeamUsers(@PathVariable String teamId, @Valid @RequestBody Upd
         team.setUsers(fromIterable(teamUsers));
         return teamRepository.save(team);
     }
+
+    // -- PRIVATE --
+
+    private void isTeamAlreadyExists(@NotNull final TeamCreateInput input) {
+        List<Team> teams = this.teamRepository.findAllByName(input.getName());
+        // 1. input is not contextual and a not contextual team already exist
+        if (FALSE.equals(input.getContextual()) && teams.stream().anyMatch(t -> FALSE.equals(t.getContextual()))) {
+            throw new UnsupportedOperationException("Global teams (non contextual) cannot have the same name (already exists)");
+        }
+        // 2. input is contextual and a contextual team on the same exercise or scenario already exist
+        if (TRUE.equals(input.getContextual())) {
+            String exerciseId = input.getExerciseIds().stream().findFirst().orElse(null);
+            if (hasText(exerciseId) && teams.stream().anyMatch(t -> TRUE.equals(t.getContextual()) && t.getExercises().stream().anyMatch((e) -> exerciseId.equals(e.getId())))) {
+                throw new UnsupportedOperationException("A contextual team with the same name already exists on this simulation");
```

Comment:
Maybe a stupid question but how come we can't have 2 teams with the same name ? Is it a rule required from product ?

Reply:
Good question, what do you think @jborozco & @EllynBsc  ?
Could we have two teams with the same name: TeamA & TeamA ?

Reply:
Hey @RomuDeuxfois & @Dimfacion 👋 
I don't see the usecase of having two teams with the same name IMO. I think differentiated naming gives structures and fosters clarity.